### ISSUE TITLE
DAS issues w/ 1.22.0-dev.6.0.

### ISSUE THREAD

**Issue by pq** [flutter/flutter-intellij#622]
This started w/ https://github.com/flutter/flutter/pull/7382 when I attempted to integrate `1.22.0-dev.5.0`.  I'd hoped it was transient or based on bad local state since neither @scheglov nor @devoncarew could repro but then @beders reported https://github.com/flutter/flutter-intellij/issues/613 which led me to believe it wasn't just me. Anyway, I'm at it again w/ https://github.com/flutter/flutter/pull/7461 which tries to pull Dart SDK `1.22.0-dev.6.0` into flutter and I get the same behavior:
* No analysis results.
* No server log output (not even a connect MSG).
* Reanalyze sources and restart server commands appear to be ignored.

**Comment by devoncarew** [comment]
This issue is also suspicious: https://github.com/flutter/flutter/pull/7439

And FYI, `1.22.0-dev.7.0` was just released.

**Comment by pq** [comment]
Some updates.

* I've confirmed that the break is introduced in `1.22.0-dev.5.0`.
* If I disable the driver (`--disable-new-analysis-driver`) in `dev.5.0`, analysis works as expected.

**Comment by pq** [comment]
Running DAS with observatory and reanalyzing sources yields this juicy exception.

```
analysis_server.dart.snapshot$main:stdout> {"error":{"code":"SERVER_ERROR","message":"NoSuchMethodError: The getter 'addedFiles' was called on null.\nReceiver: null\nTried calling: addedFiles","stackTrace":"#0      Object._noSuchMethod (dart:core-patch/object_patch.dart:44)\n#1      Object.noSuchMethod (dart:core-patch/object_patch.dart:47)\n#2      ContextManagerImpl._computeFlushedFiles (package:analysis_server/src/context_manager.dart:1059)\n#3      ContextManagerImpl._destroyContext (package:analysis_server/src/context_manager.dart:1288)\n#4      List.forEach (dart:core-patch/growable_array.dart:258)\n#5      ContextManagerImpl.refresh (package:analysis_server/src/context_manager.dart:791)\n#6      AnalysisServer.reanalyze (package:analysis_server/src/analysis_server.dart:1046)\n#7      AnalysisDomainHandler.reanalyze (package:analysis_server/src/domain_analysis.dart:278)\n#8      AnalysisDomainHandler.handleRequest (package:analysis_server/src/domain_analysis.dart:239)\n#9      AnalysisServer.handleRequest.<anonymous closure>.<anonymous closure> (package:analysis_server/src/analysis_server.dart:876)\n#10     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:189)\n#11     AnalysisServer.handleRequest.<anonymous closure> (package:analysis_server/src/analysis_server.dart:872)\n#12     _rootRun (dart:async/zone.dart:1150)\n#13     _CustomZone.run (dart:async/zone.dart:1026)\n#14     _CustomZone.runGuarded (dart:async/zone.dart:924)\n#15     runZoned (dart:async/zone.dart:1501)\n#16     AnalysisServer.handleRequest (package:analysis_server/src/analysis_server.dart:871)\n#17     ByteStreamServerChannel._readRequest.<anonymous closure> (package:analysis_server/src/channel/byte_stream_channel.dart:170)\n#18     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:189)\n#19     ByteStreamServerChannel._readRequest (package:analysis_server/src/channel/byte_stream_channel.dart:161)\n#20     ByteStreamServerChannel.listen.<anonymous closure> (package:analysis_server/src/channel/byte_stream_channel.dart:110)\n#21     _rootRunUnary (dart:async/zone.dart:1158)\n#22     _CustomZone.runUnary (dart:async/zone.dart:1037)\n#23     _CustomZone.runUnaryGuarded (dart:async/zone.dart:932)\n#24     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:331)\n#25     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:258)\n#26     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:67)\n#27     _EventSinkWrapper.add (dart:async/stream_transformers.dart:14)\n#28     _StringAdapterSink.add (dart:convert/string_conversion.dart:256)\n#29     _LineSplitterSink._addLines (dart:convert/line_splitter.dart:159)\n#30     _LineSplitterSink.addSlice (dart:convert/line_splitter.dart:134)\n#31     StringConversionSinkMixin.add (dart:convert/string_conversion.dart:180)\n#32     _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:119)\n#33     _rootRunUnary (dart:async/zone.dart:1158)\n#34     _CustomZone.runUnary (dart:async/zone.dart:1037)\n#35     _CustomZone.runUnaryGuarded (dart:async/zone.dart:932)\n#36     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:331)\n#37     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:258)\n#38     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:67)\n#39     _EventSinkWrapper.add (dart:async/stream_transformers.dart:14)\n#40     _StringAdapterSink.add (dart:convert/string_conversion.dart:256)\n#41     _StringAdapterSink.addSlice (dart:convert/string_conversion.dart:260)\n#42     _Utf8ConversionSink.addSlice (dart:convert/string_conversion.dart:336)\n#43     _Utf8ConversionSink.add (dart:convert/string_conversion.dart:329)\n#44     _ConverterStreamEventSink.add (dart:convert/chunked_conversion.dart:92)\n#45     _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:119)\n#46     _rootRunUnary (dart:async/zone.dart:1158)\n#47     _CustomZone.runUnary (dart:async/zone.dart:1037)\n#48     _CustomZone.runUnaryGuarded (dart:async/zone.dart:932)\n#49     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:331)\n#50     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:258)\n#51     _StreamController&&_SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:768)\n#52     _StreamController._add (dart:async/stream_controller.dart:635)\n#53     _StreamController.add (dart:async/stream_controller.dart:581)\n#54     _Socket._onData (dart:io-patch/socket_patch.dart:1680)\n#55     _rootRunUnary (dart:async/zone.dart:1162)\n#56     _CustomZone.runUnary (dart:async/zone.dart:1037)\n#57     _CustomZone.runUnaryGuarded (dart:async/zone.dart:932)\n#58     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:331)\n#59     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:258)\n#60     _StreamController&&_SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:768)\n#61     _StreamController._add (dart:async/stream_controller.dart:635)\n#62     _StreamController.add (dart:async/stream_controller.dart:581)\n#63     _RawSocket._RawSocket.<anonymous closure> (dart:io-patch/socket_patch.dart:1247)\n#64     _NativeSocket.issueReadEvent.issue (dart:io-patch/socket_patch.dart:781)\n#65     _microtaskLoop (dart:async/schedule_microtask.dart:41)\n#66     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)\n#67     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)\n#68     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:149)\n"},"id":"6"}
```

Now we're getting somewhere!

**Comment by scheglov** [comment]
We have a `ContextInfo` without the `AnalysisDriver` instance.

**Comment by pq** [comment]
cc @scheglov

**Comment by pq** [comment]
> We have a ContextInfo without the AnalysisDriver instance.

[Greek to me](https://en.wikipedia.org/wiki/Greek_to_me)! 😉 

Seriously though, let me know if there's anything else for me to try...

**Comment by scheglov** [comment]
Does Flutter plugin send reanalyze?
Is this stack directly after start?

**Comment by pq** [comment]
OK.  Started paused and resumed gives us:

```
Paused at isolate start (type 'continue' [F7] or 'step' [F10] to start the isolate')
$ continue
Continuing...
Unrecognized event: ServiceEvent(owner='isolates/86774781', kind='ServiceExtensionAdded', time=2017-01-12 13:27:23.490)
Unrecognized event: ServiceEvent(owner='isolates/86774781', kind='ServiceExtensionAdded', time=2017-01-12 13:27:23.490)
Unrecognized event: ServiceEvent(owner='isolates/86774781', kind='ServiceExtensionAdded', time=2017-01-12 13:27:23.666)
Unrecognized event: ServiceEvent(owner='isolates/86774781', kind='ServiceExtensionAdded', time=2017-01-12 13:27:23.666)
Isolate 452566010 'analysis_server.dart.snapshot$_cacheCleanUpFunction' has been created
analysis_server.dart.snapshot$main:stdout> {"event":"server.connected","params":{"version":"1.17.0","pid":19967,"sessionId":""}}
analysis_server.dart.snapshot$main:stdout> {"id":"0","result":{"version":"1.17.0"}}
analysis_server.dart.snapshot$main:stdout> {"id":"1"}
analysis_server.dart.snapshot$main:stdout> {"id":"2"}
analysis_server.dart.snapshot$main:stdout> {"event":"server.status","params":{"pub":{"isListingPackageDirs":true}}}
analysis_server.dart.snapshot$main:stdout> {"event":"server.status","params":{"pub":{"isListingPackageDirs":false}}}
analysis_server.dart.snapshot$main:stdout> {"error":{"code":"SERVER_ERROR","message":"NoSuchMethodError: The getter 'apiSignature' was called on null.\nReceiver: null\nTried calling: apiSignature","stackTrace":"#0      Object._noSuchMethod (dart:core-patch/object_patch.dart:44)\n#1      Object.noSuchMethod (dart:core-patch/object_patch.dart:47)\n#2      AnalysisDriver.AnalysisDriver (package:analyzer/src/dart/analysis/driver.dart:256)\n#3      ServerContextManagerCallbacks.addAnalysisDriver (package:analysis_server/src/analysis_server.dart:1827)\n#4      ContextManagerImpl._createContext (package:analysis_server/src/context_manager.dart:1186)\n#5      ContextManagerImpl._createContexts (package:analysis_server/src/context_manager.dart:1247)\n#6      ContextManagerImpl.setRoots (package:analysis_server/src/context_manager.dart:886)\n#7      AnalysisServer.setAnalysisRoots (package:analysis_server/src/analysis_server.dart:1211)\n#8      AnalysisDomainHandler.setAnalysisRoots (package:analysis_server/src/domain_analysis.dart:302)\n#9      AnalysisDomainHandler.handleRequest (package:analysis_server/src/domain_analysis.dart:241)\n#10     AnalysisServer.handleRequest.<anonymous closure>.<anonymous closure> (package:analysis_server/src/analysis_server.dart:876)\n#11     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:189)\n#12     AnalysisServer.handleRequest.<anonymous closure> (package:analysis_server/src/analysis_server.dart:872)\n#13     _rootRun (dart:async/zone.dart:1150)\n#14     _CustomZone.run (dart:async/zone.dart:1026)\n#15     _CustomZone.runGuarded (dart:async/zone.dart:924)\n#16     runZoned (dart:async/zone.dart:1501)\n#17     AnalysisServer.handleRequest (package:analysis_server/src/analysis_server.dart:871)\n#18     ByteStreamServerChannel._readRequest.<anonymous closure> (package:analysis_server/src/channel/byte_stream_channel.dart:170)\n#19     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:189)\n#20     ByteStreamServerChannel._readRequest (package:analysis_server/src/channel/byte_stream_channel.dart:161)\n#21     ByteStreamServerChannel.listen.<anonymous closure> (package:analysis_server/src/channel/byte_stream_channel.dart:110)\n#22     _rootRunUnary (dart:async/zone.dart:1158)\n#23     _CustomZone.runUnary (dart:async/zone.dart:1037)\n#24     _CustomZone.runUnaryGuarded (dart:async/zone.dart:932)\n#25     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:331)\n#26     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:258)\n#27     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:67)\n#28     _EventSinkWrapper.add (dart:async/stream_transformers.dart:14)\n#29     _StringAdapterSink.add (dart:convert/string_conversion.dart:256)\n#30     _LineSplitterSink._addLines (dart:convert/line_splitter.dart:159)\n#31     _LineSplitterSink.addSlice (dart:convert/line_splitter.dart:134)\n#32     StringConversionSinkMixin.add (dart:convert/string_conversion.dart:180)\n#33     _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:119)\n#34     _rootRunUnary (dart:async/zone.dart:1158)\n#35     _CustomZone.runUnary (dart:async/zone.dart:1037)\n#36     _CustomZone.runUnaryGuarded (dart:async/zone.dart:932)\n#37     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:331)\n#38     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:258)\n#39     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:67)\n#40     _EventSinkWrapper.add (dart:async/stream_transformers.dart:14)\n#41     _StringAdapterSink.add (dart:convert/string_conversion.dart:256)\n#42     _StringAdapterSink.addSlice (dart:convert/string_conversion.dart:260)\n#43     _Utf8ConversionSink.addSlice (dart:convert/string_conversion.dart:336)\n#44     _Utf8ConversionSink.add (dart:convert/string_conversion.dart:329)\n#45     _ConverterStreamEventSink.add (dart:convert/chunked_conversion.dart:92)\n#46     _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:119)\n#47     _rootRunUnary (dart:async/zone.dart:1158)\n#48     _CustomZone.runUnary (dart:async/zone.dart:1037)\n#49     _CustomZone.runUnaryGuarded (dart:async/zone.dart:932)\n#50     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:331)\n#51     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:258)\n#52     _StreamController&&_SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:768)\n#53     _StreamController._add (dart:async/stream_controller.dart:635)\n#54     _StreamController.add (dart:async/stream_controller.dart:581)\n#55     _Socket._onData (dart:io-patch/socket_patch.dart:1680)\n#56     _rootRunUnary (dart:async/zone.dart:1162)\n#57     _CustomZone.runUnary (dart:async/zone.dart:1037)\n#58     _CustomZone.runUnaryGuarded (dart:async/zone.dart:932)\n#59     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:331)\n#60     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:258)\n#61     _StreamController&&_SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:768)\n#62     _StreamController._add (dart:async/stream_controller.dart:635)\n#63     _StreamController.add (dart:async/stream_controller.dart:581)\n#64     _RawSocket._RawSocket.<anonymous closure> (dart:io-patch/socket_patch.dart:1247)\n#65     _NativeSocket.issueReadEvent.issue (dart:io-patch/socket_patch.dart:781)\n#66     _microtaskLoop (dart:async/schedule_microtask.dart:41)\n#67     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)\n#68     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)\n#69     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:149)\n"},"id":"3"}
analysis_server.dart.snapshot$main:stdout> {"id":"4"}
analysis_server.dart.snapshot$main:stdout> {"id":"5"}
```

**Comment by scheglov** [comment]
It looks that the `DartSdk` instance with which `AnalysisDriver` was being created does not have the summary bundle. Do you know what type of `DartSdk` this project use?

**Comment by pq** [comment]
Interesting... How would I check?

**Comment by scheglov** [comment]
There are many flavors of SDK, I thought you or @bwilkerson might know.

**Comment by bwilkerson** [comment]
@pq IntelliJ should be passing the path to the SDK on the command-line when it starts the analysis server. Does that SDK have summary (.sum) files in `lib/_internal`?

**Comment by pq** [comment]
Here's the command-line for that dart process:

```
19967 ??         0:01.51 /Users/pquitslund/src/git/clones/flutter/bin/cache/dart-sdk/bin/dart --observe --pause-isolates-on-start /Users/pquitslund/src/git/clones/flutter/bin/cache/dart-sdk/bin/snapshots/analysis_server.dart.snapshot --client-id=IntelliJ_IDEA --client-version=IC-163.10154.41 --useAnalysisHighlight2
```

`lib/_internal` contains:

```
ddc_sdk.sum		js_runtime		libraries.dart		pub			sdk_library_metadata	spec.sum		strong.sum
```

**Comment by scheglov** [comment]
If any of your projects uses `EmbedderSdk` not in strong mode, then according to [this test](https://github.com/dart-lang/sdk/blob/master/pkg/analyzer/test/src/dart/sdk/sdk_test.dart#L74), its `getLinkedBundle()` returns `null`.

**Comment by pq** [comment]
Here's the project:

https://github.com/flutter/flutter/tree/master/examples/hello_world

**Comment by scheglov** [comment]
```
Working dir: /Users/scheglov/Source/flutter/examples/hello_world
/Users/scheglov/Applications/dart-sdk/bin/pub upgrade
Resolving dependencies...
The Flutter SDK is not available.
Depended on by:
- hello_world 
Process finished with exit code 69
```

**Comment by pq** [comment]
Right.  Standard `pub` does not work for `flutter`.  You need to run `flutter packages get` or `flutter packages upgrade`.

**Comment by pq** [comment]
Alternatively I think you can just set the `FLUTTER_ROOT` env variable.

See: https://github.com/JetBrains/intellij-plugins/pull/431.

**Comment by scheglov** [comment]
Works for me.
I see that SDK is `EmbedderSdk`, and its `dartSdk.context.analysisOptions.strongMode` is `true`. So, it has the linked bundle.

Is it possible that there is an option file that turns strong mode off on your computer?

**Comment by pq** [comment]
> Is it possible that there is an option file that turns strong mode off on your computer?

This project has no `.analysis_options` at all so I would assume it would just start with the default (which is off right?) --- I suppose it could be getting options from a parent somewhere but I'm not sure.  More importantly though this is indeed the rub.  When I add options like this:

```
analyzer:
  strong-mode: true
```

and restart server, life is good!  Commenting them out and restarting gets us back to an unhappy place.

So yeah, this is the issue.  Analyzing a Flutter project w/o strong-mode enabled spells doom.

Not sure how to proceed (and I'm a bit burnt out) but at least we've identified the problem.   

Cheers!

🍻

**Comment by pq** [comment]
(FWIW: note that `flutter create` does NOT create an analysis options file so in general new flutter projects will only have default analysis options so whatever we do will have to take that into account.)

**Comment by scheglov** [comment]
Nice!
I guess it worked for me because I always opened a project in the Flutter repo, which does have `.analysis_options` file.

**Comment by pq** [comment]
A few observations now that we're out of the dark.

* DAS should fail less catastrophically in this case (cc @bwilkerson, @scheglov)
* we should consider creating `.analysis_options` in `flutter create` with `strong` flipped on and ensuring that they exist in all Flutter examples (cc @mit-mit @devoncarew)  --- in fact I thought we did this? (cc @danrubel)
* IMO this blocks integration without landing a change to disable the driver by default into the Dart SDK (cc @tvolkert)
* an avenue we might explore in a pinch is adding the VM argument programmatically in the Flutter Plugin (though this still leaves other IDEs, such as VS code broken)
* we might consider an inspection that detects this case in the IDE and encourages the user to fix it
* in general, integration tests that verify basic DAS functionality for Flutter against Dart SDK canaries would be _a good thing_ (but that's kind of a given)

More thoughts welcome!

**Comment by mit-mit** [comment]
> we should consider creating .analysis_options in flutter create 

That seems like more complexity that most users shouldn't have to care about? (See also https://github.com/flutter/flutter-intellij/issues/630). I'm expecting that prior to launch we will flip all of flutter to strong mode anyways?

**Comment by danrubel** [comment]
Once I land https://github.com/flutter/flutter/pull/7444 (which is blocked on rolling Dart SDK in flutter), then I can update `flutter create` to include an `.analysis_options` in the generated project.

**Comment by pq** [comment]
> That seems like more complexity that most users shouldn't have to care about?

You may be right.  But the rub is that until strong is not opt-in we need to ensure that all Flutter projects are strong (for the sanity of analysis) and this is a way to do that.  Or maybe that's not what you meant.  

> I'm expecting that prior to launch we will flip all of flutter to strong mode anyways?

I'm not actually sure what this means.  The Flutter platform itself is analyzed strong already.  The trouble comes when projects that use Flutter are not strong.

**Comment by mit-mit** [comment]
I really would like to avoid adding any more files to new projects. We have too many IMHO already.

If all projects need to be strong, can't we override that in code somehow rather than rely on the file? It also sounds flaky to have it in the file, as then editing it to `strong-mode: false` would cause the failure that started this thread, right?

**Comment by danrubel** [comment]
Including an `.analysis_options` file in each flutter project that points to a common `package:flutter/analysis_options_user.yaml` file accomplishes several things:
* analysis options become less magical (why does this project have these warnings and this other one does not?)
* users can easily and obviously augment the common set of flutter analysis options while continuing to inherit "best practice" analysis options settings
* flutter team can continue to improve/adjust the common set of analysis options knowing that those options will be inherited by every flutter project
* flutter analyze, dartanalyzer, and analysis server will all generate the same errors/warnings

**Comment by mit-mit** [comment]
Yes, but it violates a core UX principle of 'incremental complexity': Only expose complexity as the user needs it. 

In other words, you should need a config file to just get defaults unless we expect all users to have to edit that file. And I really don't think we will see most users configuring custom sets of analysis options, right?

**Comment by tvolkert** [comment]
@pq one clarification: is there no way to fix the underlying issue such that Stuff Works with the driver enabled?  Or is it just that doing so is sufficient work that we don't want to be blocked on it?

**Comment by bwilkerson** [comment]
> DAS should fail less catastrophically in this case ...

Absolutely. But if we cannot fall back to a non-summary solution in DAS (and that might not be feasible, and certainly isn't desirable for performance reasons), and the Flutter SDK can't have a spec.sum file in it (I'm guessing that's why we fail when strong mode is off), then we need to add some way for server to notify clients that some projects just can't be analyzed so that users can be informed.

> If all projects need to be strong, can't we override that in code somehow rather than rely on the file?

All *Flutter* projects need to be strong mode, but non-Flutter projects don't. Is there a reliable (supported long term) way for analyzer to know which projects are Flutter projects? If so, then yes, we could bake that knowledge in to analyzer.

But strong mode isn't the only setting the Flutter team wants to encourage its users to use. Recognizing that there are different groups of Flutter users, the Flutter team has put together several different sets of default options. The mechanism @danrubel was describing is basically a way for each user to specify the defaults that they want to opt in to, while still allowing them to override some options if they really want to.

**Comment by devoncarew** [comment]
We're building the sdk summaries for the flutter sdk ourselves, here:

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/cache.dart#L348
https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/dart/summary.dart#L49

**Comment by pq** [comment]
A pile of follow-ups:

@mit-mit asked:

> editing it to strong-mode: false would cause the failure that started this thread, right?

Absolutely right.  All the more reason that this situation needs to be handled more gracefully.

and @tvolkert wanted clarification:

> is there no way to fix the underlying issue such that Stuff Works with the driver enabled? Or is it just that doing so is sufficient work that we don't want to be blocked on it?

This is a call for @devoncarew (and @bwilkerson and @scheglov).  The concern I think is that the driver has had very little testing in the wild and there's worry that turning it on so late in the dev cycle might not give folks enough time to vet and make adjustments before 1.22 ships.  Their call though!

to @bwilkerson:

> the Flutter SDK can't have a spec.sum file in it (I'm guessing that's why we fail when strong mode is off)

Interestingly the `lib/_internal` directory _does_ contain a `spec.sum` (see comment above where I give the full directory listing).

(**EDIT**: just talked to @devoncarew who suggested the absence/presence of these files is a red herring.)

**Comment by devoncarew** [comment]
Right, the spec.sum and strong.sum in the sdk dir are red herrings. The embedder sdk pulls from the package dir that defines the (single) `_embedder.yaml`; in this case, sky_engine.

**Comment by devoncarew** [comment]
Update:

- a fix has landed in master
- @whesse will build a new dev sdk release
- we'll update @pq's PR with the new sdk version and the updates to create a spec mode summary in the sky_engine (_embedder.yaml) directory

**Comment by scheglov** [comment]
BTW, I've sent @pq a patch for creating both spec and strong mode Dart SDK summaries.

**Comment by pq** [comment]
PR https://github.com/flutter/flutter/pull/7461 updated.  On hold until we get a new dev SDK release with the fix for https://github.com/dart-lang/sdk/issues/28384.

**Comment by pq** [comment]
https://github.com/flutter/flutter/pull/7461 updated to dev.8.0 w/ @scheglov's fix.  Waiting on review.

**Comment by pq** [comment]
Landed.  Confirmed that this resolves the issue. 🍻 

Aside: it appears that my IDEA registry settings are no longer making their way to DAS setup.  Try as I might I now can't get the VM to start w/o isolates paused at start...  I'm sure this is an issue on my end but can't seem to get unstuck.  Any ideas welcome!
<img width="314" alt="screen shot 2017-01-17 at 6 15 56 am" src="https://cloud.githubusercontent.com/assets/67586/22023830/af3ff2fa-dc7c-11e6-8702-2f0bff2621a7.png">

(EDIT: removing `--observe` doesn't seem to do anything either?)

<img width="1361" alt="screen shot 2017-01-17 at 6 18 07 am" src="https://cloud.githubusercontent.com/assets/67586/22023852/c4c39d5c-dc7c-11e6-8c9e-3cebe6dcbb32.png">

@alexander-doroshko: any thoughts?

@scheglov: is it possible that I'm forgetting a bit I flipped elsewhere when we were debugging?  (I've got no source changes that aren't in the PR...)

**Comment by pq** [comment]
One more shot.

Registry:

<img width="821" alt="screen shot 2017-01-17 at 6 26 31 am" src="https://cloud.githubusercontent.com/assets/67586/22024202/0c196316-dc7e-11e6-9d1f-18131a126058.png">

Command run (from `ps`):

```
dart --observe --pause-isolates-on-start /Users/pquitslund/src/git/clones/flutter/bin/cache/dart-sdk/bin/snapshots/analysis_server.dart.snapshot --client-id=IntelliJ_IDEA --client-version=IC-163.10154.41 --useAnalysisHighlight2
```

The server args aren't getting there either?

**Comment by pq** [comment]
@alexander-doroshko : any chance that adding custom properties is interfering here?

<img width="353" alt="screen shot 2017-01-17 at 6 31 08 am" src="https://cloud.githubusercontent.com/assets/67586/22024379/b1c7e206-dc7e-11e6-8c55-3ea241849b01.png">

**Comment by alexander-doroshko** [comment]
I'm not aware of the influence of the custom properties or whatever else. Changing values in the registry and restarting the server should just work.

**Comment by pq** [comment]
Thanks @alexander-doroshko.  I'm due for a reboot and will let you know if the problem persists.

**Comment by pq** [comment]
Closing w/ the landing of flutter/flutter#7461.
